### ISSUE TITLE
Cannot change collection presents by a non admin user

### ISSUE THREAD

**Issue by samvasko** [directus/api#807]
# Bug Report

## Steps to Reproduce

1. Create user in a non admin without full write permissions to directus_collection_presets table
2. Attempt to resize a column as that user
3. Error is returned due to lack of permission 

## Expected Behavior
User should be able to edit `directus_collection_presets` 

## Actual Behavior
Response is 
```
{code: 302, message: "Updating item from "directus_collection_presets" collection was denied"}
```

## Other Context & Screenshots

```
2019-03-03 17:22:08] api[_].ERROR: Directus\Permissions\Exception\ForbiddenCollectionUpdateException: Updating item from "directus_collection_presets" collection was denied in /var/www/api/src/core/Directus/Permissions/Acl.php:1072
Stack trace:
#0 /var/www/api/src/core/Directus/Database/TableGateway/BaseTableGateway.php(1276): Directus\Permissions\Acl->enforceUpdateAll
#1 /var/www/api/src/core/Directus/Database/TableGateway/BaseTableGateway.php(852): Directus\Database\TableGateway\BaseTableGateway->enforceUpdatePermission
#2 /var/www/api/vendor/zendframework/zend-db/src/TableGateway/AbstractTableGateway.php(361): Directus\Database\TableGateway\BaseTableGateway->executeUpdate
#3 /var/www/api/src/core/Directus/Database/TableGateway/BaseTableGateway.php(455): Zend\Db\TableGateway\AbstractTableGateway->updateWith
#4 /var/www/api/src/core/Directus/Database/TableGateway/RelationalTableGateway.php(498): Directus\Database\TableGateway\BaseTableGateway->updateRecordByArray
#5 /var/www/api/src/core/Directus/Services/ItemsService.php(156): Directus\Database\TableGateway\RelationalTableGateway->updateRecord
#6 /var/www/api/src/core/Directus/Services/CollectionPresetsService.php(62): Directus\Services\ItemsService->update
#7 /var/www/api/src/endpoints/CollectionPresets.php(89): Directus\Services\CollectionPresetsService->update
#8 [internal function]: Directus\Api\Routes\CollectionPresets->update
#9 /var/www/api/vendor/slim/slim/Slim/Handlers/Strategies/RequestResponse.php(41): call_user_func
```

## Technical Details

Device: Desktop
OS: MacOs High Sierra
Web Server: nginx
PHP Version: 7.2
Database: Mysql 5.2
Install Method: docker 2.0.17 and 7.0.17

**Comment by samvasko** [comment]
I think this is because the `user` column in the presents db is not in the `DataTypes getUsersType` function and therefore the update cannot be correctly attributed.

**Comment by benhaynes** [comment]
Thanks @samvasko — our API Lead (@josephgagliardi) is new to the codebase, but we'll take a look and see if we can get this resolved asap!

**Comment by samvasko** [comment]
I had a look. it looks like a tough one! 👍

**Comment by rijkvanzanten** [comment]
> without full write permissions to directus_collection_presets table

The user needs write permission to `directus_collection_presets` in order to save their permissions. The error that was returned looks like expected behavior to me.

Am I misunderstanding the situation @samvasko ?

**Comment by benhaynes** [comment]
I think the expectation is to have `mine` permissions to that collection. Or we're not giving users this permission by default (so it only works for admins since they have all permissions by default). Either way, this seems like a bug that needs to be fixed.

**Comment by samvasko** [comment]
Definitely a bug. The collection is configured as mine but cannot be edited.
<img width="790" alt="Screen Shot 2019-03-11 at 22 48 37" src="https://user-images.githubusercontent.com/2674600/54163122-d51c0800-444f-11e9-9346-444d5cfe29b9.png">

**Comment by rijkvanzanten** [comment]
I think it might be the same issue that's causing https://github.com/directus/api/issues/723

**Comment by samvasko** [comment]
I did some debugging and before and It seems the same.

**Comment by bjgajjar** [comment]
@samvasko It seems the bug is fixed by #828. 
We just released API v2.0.20. Can you update to latest and let me know if it still occurs? I can not reproduce this on my side :slightly_smiling_face:
